web/src/hotkey.js

```diff
@@ -1316,31 +1418,17 @@ export function process_hotkey(e, hotkey) {
     return false;
 }
 
-/* We register both a keydown and a keypress function because
-   we want to intercept PgUp/PgDn, Esc, etc, and process them
-   as they happen on the keyboard. However, if we processed
-   letters/numbers in keydown, we wouldn't know what the case of
-   the letters were.
-
-   We want case-sensitive hotkeys (such as in the case of r vs R)
-   so we bail in .keydown if the event is a letter or number and
-   instead just let keypress go for it. */
```

Comment:
Interesting, what's the story of this comment? It seems to be explaining a reason why `keydown` events won't suffice for some of our needs.

Have you confirmed that the thing it warns about seems to still work fine? Have you been able to work out what changed since this comment was written?

Reply:
> However, if we processed letters/numbers in keydown, we wouldn't know what the case of the letters were.

Let's take `r` vs `R` as an example.

Earlier, we were using `event.which`.

For `r`:
| Event type | which |
| --- | --- |
| keypress | 114 |
| keydown | 82 |

For `R`:
| Event type | which |
| --- | --- |
| keypress | 82 |
| keydown | 82 |

So, if we use `event.which` with `keydown`, the value is `82` for both `r` and `R`. No way to distinguish. So, at previously we were taking help of `keypress` -- `114` vs `82`.

But now we're using `.key` which clearly has the value `r` vs `R` in `keydown`. So, we can safely stop relying on `.keypress`.

Reply:
> Have you confirmed that the thing it warns about seems to still work fine?

I have manually tested few of the hotkeys which were earlier managed by `keypress` and now managed by `keydown`; they work as expected. E.g. `a` vs `A`

> Have you been able to work out what changed since this comment was written?

`.which` is now legacy. `.key` is the thing now supported by major browsers -- https://caniuse.com/?search=KeyboardEvent.key

Btw, the finalized version of the spec was released just ~2 weeks ago (https://www.w3.org/TR/uievents-key/). But majority of the browsers have already implemented the draft specs, so I think we're safe to use it.

Reply:
Let's copy some of this explanation into the commit message for posterity.

Reply:
Great, thanks for that explanation.

It sounds like the direct answer to "what changed since this comment was written" is: we switched from `.which` to `.key`, in one of the earlier commits in this very PR. :-) So drawing that connection will be one key piece of information to put into the commit message.

Reply:
Updated the commit message.